crates/mpz-ole-core/src/ideal/role.rs

```diff
@@ -0,0 +1,94 @@
+//! This module provides an ideal ROLE functionality.
```

Comment:
```suggestion
//! Ideal ROLE.
```

Style wise, I think I prefer very succinct module and type headlines. It just reads cleaner with rustdoc to avoid leading words like "this module provides..." etc. Of course, its totally fine to be more verbose in the following description